vendor/github.com/ethereum/go-ethereum/core/headerchain.go

```diff
@@ -271,6 +292,11 @@ func (hc *HeaderChain) InsertHeaderChain(chain []*types.Header, writeHeader WhCa
 			stats.ignored++
 			continue
 		}
+
+		// introduce delay for the expensive write to
+		// trade off sync time with CPU usage
+		time.Sleep(writeDelay)
+
 		if err := writeHeader(header); err != nil {
```

Comment:
Is this method sync or async? Does it try to do IO operation directly?

Reply:
I think it's synchronous, that's why introducing delay has an effect.